sklearn/metrics/pairwise.py

```diff
@@ -935,13 +939,46 @@ def _parallel_pairwise(X, Y, func, n_jobs, **kwds):
     if Y is None:
         Y = X
 
+    if n_jobs == 1:
```

Comment:
delayed does a pre-check... which can be turned off, mind you, but I thought it would be clearer just to pre-empt.

Also, handling `n_jobs==1` like this avoids the copy in `hstack` which could be costly for a large distance matrix.